apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -768,19 +762,51 @@ export async function updateLti13Scores(
     }),
   );
 
+  const course_staff_data = await selectUsersWithCourseInstanceAccess({
+    course_instance_id: assessment.course_instance_id,
+    minimal_role: 'Student Data Viewer',
+  });
+  const course_staff = new Set(course_staff_data.map((staff) => staff.uid));
+
   const memberships = await Lti13ContextMembership.loadForInstance(instance);
+
+  const token = await getAccessToken(instance.lti13_instance.id);
```

Comment:
In the worst case, this token is only valid for five minutes. Are we pretty confident that grade syncing will take less than that even for large classes? If not, we might consider doing this in the loop too.

Reply:
Good catch. I moved it into the loop in 725e64af39